jscomp/core/lam_compile_primitive.ml

```diff
@@ -36,8 +36,47 @@ let ensure_value_unit (st : Lam_compile_context.continuation) e : E.t =
   | EffectCall Not_tail -> e
 (* NeedValue should return a meaningful expression*)
 
-let translate loc (cxt : Lam_compile_context.t) (prim : Lam_primitive.t)
-    (args : J.expression list) : J.expression =
+let rec module_of_expression = function
+  | J.Var (J.Qualified (module_id, value)) -> [ (module_id, value) ]
+  | J.Caml_block (exprs, _, _, _) ->
+      exprs
+      |> List.map (fun (e : J.expression) ->
+             module_of_expression e.expression_desc)
+      |> List.concat
+  | _ -> []
+
+let get_module_system () =
+  let packages_info = Js_packages_state.get_packages_info () in
+  let module_systems =
+    Js_packages_info.map packages_info (fun { module_system } -> module_system)
+  in
+  match module_systems with
+  | [] -> assert false
```

Comment:
Looking at the code, there's test mode where the module system is empty. Not sure what this implies in practice.

Reply:
Yes, when I print the dump lambda IR, the module system seems empty.